pkg/cmd/server/api/types.go

```diff
@@ -417,6 +433,28 @@ type ImagePolicyConfig struct {
 	// MaxScheduledImageImportsPerMinute is the maximum number of image streams that will be imported in the background per minute.
 	// The default value is 60. Set to -1 for unlimited.
 	MaxScheduledImageImportsPerMinute int
+	// AllowedRegistriesForImport is a white list of registries that we allow to import the
+	// images from.
+	// When a new master config is being generated, several public registries like
+	// docker.io, Red Hat registry, gcr.io, Amazon ECS, quay.io are allowed by default.
+	// If you want to import images from your private Docker Registry, you have to add it into
+	// the list.
+	// If not specified in master config all registries are allowed by default (legacy
+	// behavior).
+	AllowedRegistriesForImport *AllowedRegistries
+}
+
+// AllowedRegistries represents a list of registries allowed for the image import.
+type AllowedRegistries []RegistryLocation
+
+// RegistryLocation contains a location of the registry specified by the registry domain
+// name. The domain name might include wildcards, like '*' or '??'.
+type RegistryLocation struct {
+	// DomainName specifies a domain name for the registry
+	DomainName string
+	// Insecure indicates whether the registry is secury (https) or insecure (http)
+	// By default (if not specified) the registry is assumed as secure.
+	Insecure bool
```

Comment:
If it's part of domain name (I assume from validation code) can you state that in the doc text for `DomainName`, it isn't obvious.